docs/dsl1.rst

```diff
@@ -0,0 +1,192 @@
+
+    process splitLetters {
+
+        output:
+        file 'chunk_*' into letters
+
+        """
+        printf '${params.str}' | split -b 6 - chunk_
+        """
+    }
+
+    process convertToUpper {
+
+        input:
+        file x from letters.flatten()
+
+        output:
+        stdout result
+
+        """
+        cat $x | tr '[a-z]' '[A-Z]'
+        """
+    }
+
+    result.view { it.trim() }
+
+To migrate this code to DSL2, you need to move all of your channel logic throughout the script
+into a (nameless) ``workflow`` definition. Additionally, you must call each process explicitly,
+passing any input channels as arguments (instead of ``from ...``) and receiving any output channels
+as return values (instead of ``into ...``).
+
+Refer to the :ref:`workflow-page` page to learn how to define a workflow, as well as the original
+:ref:`getstarted-first` example to see the resulting DSL2 version.
+
+
+Channel forking
+===============
+
+In DSL1, a channel can be used as an input only once; to use a channel multiple times, the channel must
+be forked using the ``into`` operator.
+
+In DSL2, channels are automatically forked when connecting two or more consumers.
+
+For example::
+
+    channel
+        .from('Hello','Hola','Ciao')
+        .set{ cheers }
+
+    cheers
+        .map{ it.toUpperCase() }
+        .view()
+
+    cheers
+        .map{ it.reverse() }
+        .view()
+
+Similarly, process outputs can be consumed by multiple consumers automatically, which makes workflow scripts
+much easier to read and write.
+
+
+Modules
+=======
+
+In DSL1, the entire Nextflow pipeline must be defined in a single file (e.g. ``main.nf``). This
+restriction becomes quite cumbersome as a pipeline becomes larger, and it hinders the sharing and
+reuse of pipeline components.
+
+DSL2 introduces the concept of "module scripts" (or "modules" for short), which are Nextflow scripts
+that can be "included" by other scripts. While modules are not essential to migrating to DSL2, nor are
+they mandatory in DSL2 by any means, modules can help you organize a large pipeline into multiple smaller
+files, and take advantage of modules created by others. Check out the :ref:`module-page` to get started.
+
+
+Deprecations
+============
+
+Processes
+---------
+
+* The ``set`` process input type is no longer supported, use :ref:`tuple <process-input-tuple>` instead.
+* The ``set`` process output type is no longer supported, use :ref:`tuple <process-out-tuple>` instead.
+* The ``mode flatten`` option for process outputs is no longer available. Use the :ref:`operator-flatten` operator on the corresponding output channel instead.
+
+* Unqualified value and file elements in a tuple declaration are no longer allowed. Use an explicit
+  ``val`` or ``path`` qualifier.
+  
+  For example::
+
+    process foo {
+        input:
+        tuple X, 'some-file.sam'
+        output:
+        tuple X, 'some-file.bam'
+
+        script:
+        '''
```

Comment:
I think that these need to be double quotes for this example to work, no?

Reply:
You are right, thanks